feat: added UI for importing ERC20 tokens

impelcrypto opened the pull request:
**Pull Request Summary**

* feat: added UI for importing ERC20 tokens

**Check list**
- [ ] contains breaking changes
- [ ] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**

* feat: added an option menu on the assets panel
  * style: users can hide the small balances tokens from the assets panel
  * style: users can open the `Import tokens` modal 
* feat: added a modal for importing ERC20 tokens

![image](https://user-images.githubusercontent.com/92044428/174056000-37f0e08e-0289-4aed-9d73-4acb81ebfe51.png)

![image](https://user-images.githubusercontent.com/92044428/174056098-4673bb51-d846-4994-b413-f232564b91f1.png)



**To-dos**
* add new icons into Astar-UI library

sirius651 commented:
![스크린샷 2022-06-20 오후 2 43 15](https://user-images.githubusercontent.com/1606820/174532925-55cc7acc-e779-41f0-8820-ef6397df2a42.png)

found out tiny thing other than that great. when I click on `Add to wallet`, then show metamask popup. the image of token is crashed.

Kahonnohak commented:
Lovely experience!

Could you move those contents to the right? We can have "bridge" and make it disabled.

Why we don't see the token icons?

<img width="561" alt="Screenshot 2022-06-20 at 14 55 08" src="https://user-images.githubusercontent.com/77480847/174534669-96ec5dd3-ed45-486d-a641-28bd06af1666.png">

impelcrypto commented:
@Kahonnohak 
> Why can't we see the token icons?

The app doesn't have the image source of token icons.

Kahonnohak commented:
> @Kahonnohak
> 
> > Why can't we see the token icons?
> 
> The app doesn't have the image source of token icons.

The app = Portal?

Metamask have ARSW token icon. Could we bring this to Portal? 

Different topic but can we have ASTR token icon on Metamask?

<img width="355" alt="Screenshot 2022-06-20 at 15 10 31" src="https://user-images.githubusercontent.com/77480847/174536497-02aaabd4-43a2-470e-a4d9-ff993f2b1674.png">

impelcrypto commented:
@Kahonnohak 

Yes, the app means portal. To achieve this, the app has to provide the image to MetaMask whenever users import tokens. The problem is the app doesn't know how to get and provide the token source to MetaMask extention unless the source has been hardcoded into the app. 
Perhaps you can try importing any tokens on MetaMask directly. MetaMask won't display the token icon because MetaMask doesn't know what is the icon for the imported token.

Kahonnohak reviewed (approved):
THANK YOU!! Thanks for adding token symbols too!! (I did understand why we can't)

impelcrypto commented:
@Kahonnohak 

> https://github.com/AstarNetwork/astar-apps/pull/445#issuecomment-1160006144
> https://github.com/AstarNetwork/astar-apps/pull/445#issuecomment-1160016573

Fixed. By the way, I have added a key-value dataset to store the image source of the custom tokens :D

![image](https://user-images.githubusercontent.com/92044428/174609111-70a26abf-a20f-4f9e-95ba-da0de4e3bf90.png)

![image](https://user-images.githubusercontent.com/92044428/174609144-9b6d722d-39b7-4486-b917-07a8e7152633.png)
